tests/driver/index.ts

```diff
@@ -0,0 +1,438 @@
+
+          // Prefer calling enable over editing trunk.yaml directly because it also handles runtimes, etc.
+          const enableCommand = `${
+            this.inputArgs.cliPath ?? "trunk"
+          } check enable ${linterVersionString} --monitor=false`;
+          execSync(enableCommand, { cwd: this.sandboxPath });
+        } catch (error) {
+          console.warn(`Failed to enable ${this.linter} with message ${(error as Error).message}`);
+        }
+      }
+    }
+  }
+
+  // Delete the sandbox testing directory and its contents, as well as removing any trunk information
+  // associated with it in order to prune the cache.
+  async TearDown() {
+    const deinitCommand = `${this.inputArgs.cliPath ?? "trunk"} deinit`;
+    exec(deinitCommand, { cwd: this.sandboxPath, timeout: 1000 });
+    if (this.sandboxPath) {
+      fs.rmSync(this.sandboxPath, { recursive: true });
+    }
+  }
+
+  // Run a specified trunk command with 'args' (omit 'trunk') and additional options.
+  async Run(args: string, execOptions?: ExecOptions) {
+    const fullCommand = `${this.inputArgs.cliPath ?? "trunk"} ${args}`;
+    return execpromise(fullCommand, { cwd: this.sandboxPath, ...execOptions });
+  }
+
+  // Run a 'trunk check' command with additional options.
+  async RunCheck(
+    args = "",
+    targetAbsPath?: string,
+    resultJsonPath: string = path.join(this.sandboxPath ?? "", "result.json")
+  ) {
+    const fullArgs = `check -n --output-file=${resultJsonPath} --no-progress ${args}`;
+
+    return this.Run(fullArgs)
+      .then(({ stdout, stderr }) =>
+        this.ParseRunResult(
+          {
+            exitCode: 0,
+            stdout,
+            stderr,
+            outputJson: JSON.parse(fs.readFileSync(resultJsonPath, { encoding: "utf-8" })),
+          },
+          ITrunkVerb.Check,
+          targetAbsPath
+        )
+      )
+      .catch((error: Error) => {
+        const er = error as any;
+        const trunkRunResult = <ITrunkRunResult>{
+          exitCode: er["code"],
+          stdout: er["stdout"],
+          stderr: er["stderr"],
+          outputJson: JSON.parse(fs.readFileSync(resultJsonPath, { encoding: "utf-8" })),
+        };
+
+        if (trunkRunResult.exitCode != 1) {
+          console.log("Failure running 'trunk check'");
+          console.log(error);
+        }
+        return this.ParseRunResult(trunkRunResult, ITrunkVerb.Check, targetAbsPath);
+      });
+  }
+
+  // Run 'trunk check' on a specified target 'targetRelativePath' with a 'linter' filter.
+  // Prefer this to 'RunCheck' when possible.
+  async RunCheckUnit(targetRelativePath: string, linter: string): Promise<ITestResult> {
+    const targetAbsPath = path.join(this.sandboxPath ?? "", targetRelativePath);
+    const resultJsonPath = `${targetAbsPath}.json`;
+    const args = `--upstream=false --filter=${linter} ${targetAbsPath}`;
+    return this.RunCheck(args, targetAbsPath, resultJsonPath);
```

Comment:
nit: traces are a lot easier to follow i f you `await` at each function in the stack

```suggestion
    return await this.RunCheck(args, targetAbsPath, resultJsonPath);
```